### DISCUSSION TITLE
querying BMW sever vs. the BMW car

### DISCUSSION THREAD

**Discussion by floriangit** [bimmerconnected/bimmer_connected#508] [Q&A] [answered]
Happy New Year! And thanks for this app, it worked out-of-the-box with my new U06 PHEV :)

When I query with bimmerconnect on CLI, is it OK to query say every minute?
I'm afraid when I query often, that also my BMW wakes up more often....? Or does the app only query the BMW server and this never results in a query to the car?

Thanks.

**Comment by rikroe** [comment, marked as answer]
Happy new year to you too!

We only query the data on BMW servers. Depending on your car and infotaiment system, your car updates the BMW servers either at a regular interval when active (there seem to be some cars/regions where even the car location is updated while moving) or at least when the car is stopped/locked/ignition turned off.

However calling every minute will give you issues. We are currently rate-limited by BMW to poll only every 20-30 minutes (30 minutes should be safe). Check https://github.com/home-assistant/core/issues/83128#issuecomment-1355313924 for any updates on this matter.

**Reply by floriangit** [reply]
Thanks for your kind reply. I understand and will try to keep informed, no problem!
I would add that (it seems to me!) not only the car updates the servers, but also vice versa. Sometimes using your app it's quite clear that querying the BMW server first checks with the car, noticed by the network exchange timing taking 1-3 seconds before I get the (XML? JSON?) output.
Anyhow, thanks for bimmer_connected, I'll try to make best use of it :-) Regarding BMW IT, I work on my day2day job (software tier1 to BMW) with it, and I find it's "evolving" (it was worse before, but it seems to get better, lol).
